### PR TITLE
keep the `STAGE0_MISSING_TARGETS` list updated

### PR DESCRIPTION
**Pull request by onur-ozkan**
Implements https://github.com/rust-lang/rust/pull/124461#issuecomment-2092574309.

r? pietroalbini

### CODE REVIEW

**Comment by onur-ozkan** [comment]
due to inactivity, r? bootstrap

**Comment by pietroalbini** [comment]
r? @pietroalbini 

Was at a conference and then on vacation.

File: src/bootstrap/src/core/sanity.rs [hunk 1]
```diff
@@ -195,8 +193,22 @@ than building it.
         if !["A", "B", "C"].contains(&target_str.as_str()) {
             let mut has_target = false;
 
-            let supported_target_list =
-                output(Command::new(&build.config.initial_rustc).args(["--print", "target-list"]));
+            let supported_target_list: HashSet<String> =
+                output(Command::new(&build.config.initial_rustc).args(["--print", "target-list"]))
```
Comment (pietroalbini) [inline src/bootstrap/src/core/sanity.rs:197]: Noting that we are doing this inside of a for loop, while we only need to execute it once. Invoking rustc multiple times could have non-trivial impact on bootstrap startup time if multiple targets are passed.

**Review by emilyalbini — CHANGES_REQUESTED** [review]

File: src/bootstrap/src/core/sanity.rs (hunk 1 above)
Comment (onur-ozkan) [reply src/bootstrap/src/core/sanity.rs:197]: Right

**Review by emilyalbini — APPROVED** [review]

**Pull request merged**